syscalls_darwin.go

```diff
@@ -75,6 +75,15 @@ func openDevSystem(config Config) (ifce *Interface, err error) {
 	if config.DeviceType != TUN {
 		return nil, errors.New("only tun is implemented for SystemDriver, use TunTapOSXDriver for tap")
 	}
+
+	ifIndex := -1
+	if config.Name != "" {
+		_, err := fmt.Sscanf(config.Name, "utun%d", &ifIndex)
+		if err != nil || ifIndex < 0 {
+			return nil, fmt.Errorf("Interface name must be utun[0-9]+")
+		}
```

Comment:
Sorry just one more thing. This would cause parsing `utun20abc` into `utun20`. It'd be nice if that can be rejected. Perhaps something like this?

```suggestion
		const utunPrefix = "utun"
		if !strings.HasPrefix(config.Name, utunPrefix) {
			return nil, fmt.Errorf("Interface name must be utun[0-9]+")
		}
		ifIndex, err := strconv.Atoi(config.Name[len(utunPrefix):])
		if err != nil || ifIndex < 0 || ifIndex > math.MaxUint32 - 1 {
			return nil, fmt.Errorf("Interface name must be utun[0-9]+")
		}
```